doc: Import support documentation and considerations (#1305)
mongodb/mongodbatlas-cloudformation-resources: cfn-resources/README.md

| 47  | stream-connection                                        | ![Build](https://img.shields.io/badge/GA-green) | [example](../examples/atlas-streams/stream-connection/stream-connection.json)                                                                                      | [./stream-connection/test](./stream-connection/test)  
|47  | resource-policy                                        | ![Build](https://img.shields.io/badge/Beta-yellow) | [example](../examples/resource-policy/resource-policy.json)                                                                                      | [./resource-policy/test](./resource-policy/test)  

> ## Resource Import Operations

Reviewer: let's have a section to call out (if any) what resources don't currently support Import due to limitations

Author: see other [reply](https://github.com/mongodb/mongodbatlas-cloudformation-resources/pull/1305#discussion_r1989217831). All currently published resources have the update operation available so there are no resources that can't be imported. Do you think it's still necessary to add the section?

Author: Changed in [df778ac](https://github.com/mongodb/mongodbatlas-cloudformation-resources/pull/1305/commits/df778ac7f796d417f3f4e906d253c0ef691dfa98). Let me know if now it's okay

MongoDB Atlas AWS CloudFormation resources support the import operation, allowing you to bring existing Atlas resources under CloudFormation management. When importing resources, please consider the following:

> ### Import Requirements and Considerations

Reviewer: shouldn't we say that Update operation needs to be there because an Update is called?

Author: I don't think it's necessary since all the resources that we have published have full CRUD support so all of them have update operation implemented

Reviewer: got it - but I think we can still mention it as a consideration. It's something that I wouldn't personally expect to happen (an Update during an Import)

Author: Done in [df778ac](https://github.com/mongodb/mongodbatlas-cloudformation-resources/pull/1305/commits/df778ac7f796d417f3f4e906d253c0ef691dfa98).

| 47  | stream-connection                                        | ![Build](https://img.shields.io/badge/GA-green) | [example](../examples/atlas-streams/stream-connection/stream-connection.json)                                                                                      | [./stream-connection/test](./stream-connection/test)  
|47  | resource-policy                                        | ![Build](https://img.shields.io/badge/Beta-yellow) | [example](../examples/resource-policy/resource-policy.json)                                                                                      | [./resource-policy/test](./resource-policy/test)  

## Resource Import Operations

> All MongoDB Atlas AWS CloudFormation resources support the import operation, allowing you to bring existing Atlas resources under CloudFormation management. When importing resources, please consider the following:

Author: Addressed, thanks!

> All MongoDB Atlas AWS CloudFormation resources support the import operation, allowing you to bring existing Atlas resources under CloudFormation management. When importing resources, please consider the following:

Reviewer: @oarbusi should we also mention that Import is available in the [main readme](https://github.com/mongodb/mongodbatlas-cloudformation-resources?tab=readme-ov-file) and link to here?

> All MongoDB Atlas AWS CloudFormation resources support the import operation, allowing you to bring existing Atlas resources under CloudFormation management. When importing resources, please consider the following:

Author: done in [a514b82](https://github.com/mongodb/mongodbatlas-cloudformation-resources/pull/1305/commits/a514b829c2238d4e97ca88bd3901cff56831b149)

### Import Requirements and Considerations

1. **DeletionPolicy Attribute**: Your resource template must include the [DeletionPolicy](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-attribute-deletionpolicy.html) attribute in order to specify what happens to the resource when it is removed from the stack.

> 2. **UPDATE Operation Required**: The import operation executes the UPDATE operation behind the scenes. Therefore, UPDATE functionality must be properly implemented for the resource type to support import.

Reviewer: thank you, the nit I have is that this should be linked to some AWS documentation or at least specify that it's how AWS CFN works

> 2. **UPDATE Operation Required**: The import operation executes the UPDATE operation behind the scenes. Therefore, UPDATE functionality must be properly implemented for the resource type to support import. For more datails on when the update is run, please see [Create a stack from existing resources using the AWS Management Console](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/resource-import-new-stack.html#resource-import-new-stack-console) or [Import an existing resource into a stack using the AWS Management Console](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/resource-import-existing-stack.html#resource-import-existing-stack-console) or 

Reviewer 2: ```suggestion
2. **UPDATE Operation Required**: The import operation executes the UPDATE operation behind the scenes. Therefore, you must properly implement UPDATE functionality for the resource type to support import. For more datails on when the UPDATE is run, please see [Create a stack from existing resources using the AWS Management Console](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/resource-import-new-stack.html#resource-import-new-stack-console) or [Import an existing resource into a stack using the AWS Management Console](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/resource-import-existing-stack.html#resource-import-existing-stack-console) or 
```

> 3. **Outputs Restrictions**: As part of the import operation, you cannot modify or add [Outputs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/outputs-section-structure.html) to the template. However, you can add outputs after the import is complete.

Reviewer 2: ```suggestion
3. **Outputs Restrictions**: You cannot modify or add [Outputs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/outputs-section-structure.html) to the template during the import operation. However, you can add outputs after the import is complete.
```
